src/test/kotlin/se/uulm/snowballr/backend/service/user/RegisterTest.kt

```diff
@@ -1,86 +1,109 @@
 import io.mockk.every
-import io.mockk.mockk
 import kotlinx.coroutines.DelicateCoroutinesApi
 import kotlinx.coroutines.ExperimentalCoroutinesApi
-import org.junit.jupiter.api.AfterEach
-import org.junit.jupiter.api.BeforeEach
 import org.junit.jupiter.api.Test
 import org.junit.jupiter.api.assertDoesNotThrow
 import org.junit.jupiter.api.assertThrows
-import org.koin.core.context.startKoin
-import org.koin.core.context.stopKoin
-import org.koin.dsl.module
-import org.koin.test.KoinTest
 import se.uulm.snowballr.backend.DataBuilder
-import se.uulm.snowballr.backend.RandomKeyGenerator
-import se.uulm.snowballr.backend.env.EnvReader
+import se.uulm.snowballr.backend.auth.GrpcContext
 import se.uulm.snowballr.backend.model.SnowballRException
+import se.uulm.snowballr.backend.model.jwt.JwtTokens
 import se.uulm.snowballr.backend.service.MainServiceTest
 import se.uulm.snowballr.backend.testCoroutine
 import snowballr.Authentication
+import kotlin.test.assertEquals
 
 @ExperimentalCoroutinesApi
 @DelicateCoroutinesApi
-class RegisterTest : MainServiceTest(), KoinTest {
-    private val envReaderMock = mockk<EnvReader>()
-
-    @BeforeEach
-    override fun setUpTest() {
-        super.setUpTest()
-        // Stop any existing Koin context
-        stopKoin()
-
-        // Start Koin context with a mock module
-        startKoin {
-            modules(
-                module {
-                    single { envReaderMock }
-                },
-            )
-        }
-
-        // Mock JWT key pair
-        val (privateKeyBase64, publicKeyBase64) = RandomKeyGenerator.generateKeyPair()
-        every { envReaderMock.env.encryption.jwtPrivateKeyBase64 } returns privateKeyBase64
-        every { envReaderMock.env.encryption.jwtPublicKeyBase64 } returns publicKeyBase64
-    }
+class RegisterTest : MainServiceTest() {
+    @Test
+    fun `When a user provides valid credentials, the user is registered successfully`() = testCoroutine {
+        val cookiesMap = mutableMapOf<String, String?>()
 
-    @AfterEach
-    override fun tearDownTest() {
-        stopKoin()
-        super.tearDownTest()
-    }
+        // Create a new context with cookiesMap and run code inside it
+        val initialContext = Context.current()
+            .withValue(GrpcContext.COOKIES_TO_SET_CONTEXT_KEY, cookiesMap)
+        initialContext.attach()
 
-    @Test
-    fun `When a user is registered, then no exception is thrown`() = testCoroutine {
-        val request = Authentication.RegisterRequest.newBuilder().build()
-        val user = DataBuilder.createExampleUser()
+        val testUser = DataBuilder.createExampleUser()
+        val userPassword = "AAbb__00"
+        val tokens = JwtTokens("accessToken", "refreshToken")
+
+        val request = Authentication.RegisterRequest.newBuilder().apply {
+            email = testUser.email
+            password = userPassword
+            firstName = testUser.firstName
+            lastName = testUser.lastName
+        }.build()
```

Comment:
Do you really need to build a complete request? You mock almost everything except for the password hashing
```suggestion
        val request = Authentication.RegisterRequest.newBuilder()
            .setPassword("AAbb__00")
            .build()
```

Reply:
You're right. All of this can be removed.